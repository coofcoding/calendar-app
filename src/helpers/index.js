export * from './CalendarLocalizer';
export * from './getEnvVariables';
export * from './DateEventsParse';