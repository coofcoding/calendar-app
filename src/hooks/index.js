export * from './useForm';
export * from './useEvent';
export * from './useUiStore';
export * from './useCalendarStore';