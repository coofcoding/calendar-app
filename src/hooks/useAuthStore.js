import { useSelector } from "react-redux";
import { useDispatch } from "react-redux";
import { calendarApi } from "./../api";
import { checking, clearErrorMessage, onLogin, onLogout } from "../store";

export const useAuthStore = () => {
  const { status, user, errorMessage } = useSelector(state => state.auth);
  const dispatch = useDispatch();

  const startLogin = async ({ email, password }) => {
    console.log({ email, password });

    try {
      dispatch(checking());
      const { data } = await calendarApi.post("/auth", { email, password });

      localStorage.setItem("token", data.token);
      localStorage.setItem("token-init-date", new Date().getTime());

      dispatch(onLogin({ name: data.name, uid: data.uid }));
    } catch (error) {
      dispatch(onLogout("Incorrect credentials"));
      setTimeout(() => {
        dispatch(clearErrorMessage());
      }, 1000);
    }
  };

  const startRegister = async ({ name, email, password }) => {
    try {
      dispatch(checking());
      const { data } = await calendarApi.post("/auth/new", {
        name,
        email,
        password,
      });

      localStorage.setItem("token", data.token);
      localStorage.setItem("token-init-date", new Date().getTime());

      dispatch(onLogin({ name: data.name, uid: data.uid }));
    } catch (error) {
      !!error.response.data.msg && dispatch(onLogout(error.response.data.msg));
      setTimeout(() => {
        dispatch(clearErrorMessage());
      }, 1000);
    }
  };

  const checkAuthToken = async () => {
    const token = localStorage.getItem("token");

    if (!token) return dispatch(onLogout());

    try {
      const { data } = await calendarApi.get("auth/renew");
      localStorage.setItem("token", data.token);
      localStorage.setItem("token-init-date", new Date().getTime());
      dispatch(onLogin({ name: data.name, uid: data.uid }));
    } catch (error) {
      localStorage.clear();
      dispatch(onLogout());
    }
  };

  const startLogout = () => {
    localStorage.clear();
    dispatch( onLogout() );
  }

  return {
    //* Properties
    status,
    user,
    errorMessage,

    //* Methods
    startLogin,
    startRegister,
    checkAuthToken,
    startLogout
  };
};
