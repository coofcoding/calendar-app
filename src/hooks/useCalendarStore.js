import { useSelector, useDispatch } from 'react-redux';
import { onAddNewEvent, onDeleteEvent, onSetActiveEvent, onUpdateEvent } from '../store';

export const useCalendarStore = () => {

    const dispatch = useDispatch();
    const { events, activeEvent } = useSelector( state => state.calendar );

    const setActiveEvent = ( calendarEvent ) => {
        dispatch( onSetActiveEvent(calendarEvent) )
    }

    const startSavingEvent = async ( calendarEvent ) => {
        
        if( calendarEvent._id )  {
            // Editing an existing event
            dispatch( onUpdateEvent({ ...calendarEvent }));
        } else {
            // Creating new event
            dispatch( onAddNewEvent({ ...calendarEvent, _id: new Date().getTime() }) )
        }
    }

    const startDeleteEvent = () => {
        dispatch( onDeleteEvent() );
    }

    return {
        //* Properties
        events,
        activeEvent,

        //* Methods
        setActiveEvent,
        startSavingEvent,
        startDeleteEvent
    }

}