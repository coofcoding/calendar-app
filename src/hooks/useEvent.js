import { addHours, differenceInSeconds } from "date-fns";
import { useMemo } from "react";
import { useState } from "react";
import Swal from 'sweetalert2';

export const useEvent = () => {
    
    const [isOpen, setIsOpen] = useState(true);
    const [formSubmitted, setFormSubmitted] = useState(false);

    const [formValues, setFormValues] = useState({
        title: '',
        notes: '',
        start: new Date(),
        end: addHours( new Date(), 2),
    })

    const titleClass = useMemo(() => {
        if ( !formSubmitted ) return '';
        console.log('check')
        return ( formValues.title.trim().length <= 0 ) ? 'border-red-400 bg-red-100 placeholder:text-red-400 focus:text-slate-500' : ''
    }, [formValues.title, formSubmitted]) 

    const handleChange = ({target}) => {
        setFormValues({
            ...formValues,
            [target.name]: target.value
        })
    }

    const handleDateChange = ( event, changing ) => {
        setFormValues({
            ...formValues,
            [changing]: event
        })
    }

    const onCloseModal = () => {
        setIsOpen(false)
    }

    const handleSubmit = ( event ) => {
        event.preventDefault();
        setFormSubmitted(true);

        const differente = differenceInSeconds( formValues.end, formValues.start );

        if ( isNaN(differente) || differente < 1 ){
            Swal.fire({
                title: 'Error in data',
                text: 'Please check the date and time and try again.',
                icon: 'error',
                showConfirmButton: false,
                timer: 3000
            });
            return;
        }

        if ( formValues.title.trim().length <= 0 ) return;

    }

    return {
        formValues,
        handleChange, 
        handleDateChange,
        handleSubmit,
        isOpen, 
        onCloseModal,
        titleClass, 
    }
}