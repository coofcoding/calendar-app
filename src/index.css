@tailwind base;
@tailwind components;
@tailwind utilities;

body {
  font-family: 'Poppins', sans-serif;
}

/* Modal */
.ReactModalPortal > div{
  opacity: 0;
}

.ReactModalPortal .ReactModal__Overlay {
  align-items: center;
  display: flex;
  justify-content: center;
  transition: opacity .2s ease-in-out;
  z-index: 999;
}

.modal-fondo {
  background-color: rgba(0, 0, 0, 0.3);
  bottom: 0;
  left: 0;
  right: 0;
  top: 0;
  position: fixed;
}

.ReactModalPortal .ReactModal__Overlay--after-open {
  opacity: 1;
}

.ReactModalPortal .ReactModal__Overlay--before-close {
  opacity: 0;
}

::-webkit-scrollbar {
  width: 5px;
}

::-webkit-scrollbar-track {
  border-radius: 10px;
}

::-webkit-scrollbar-thumb {
  background-clip: content-box;
  border: 5px solid transparent !important;
  background: #a4a4a4;
  border-radius: 50px;
}

.react-datepicker {
  font-family: 'Poppins', sans-serif !important;
  border: none !important;
  border-radius: 1rem !important;
  background: transparent !important;
}

.react-datepicker__month-container {
  border: 1px solid #d8d8d8 !important;
  border-radius: 17px;
  background-color: white;
}

.react-datepicker__header:not(.react-datepicker__header--has-time-select) {
  border-top-right-radius: 17px !important;
}

.react-datepicker__header {
  padding: 10px !important;
  color: #a4a4a4 !important;
  border-top-left-radius: 17px !important;
  border-top-right-radius: 17px !important;
  border: none !important;
}

.react-datepicker-popper[data-placement^=bottom] .react-datepicker__triangle {
  display: none;
}

.react-datepicker__navigation--previous {
  top: 10px !important;
  left: 10px !important;
}

.react-datepicker__navigation--next {
  top: 10px !important;
  right: 93px !important;
}

.react-datepicker__day {
  color: #6b6b6b !important;
}

.react-datepicker__day-name {
  color: #a5a5a5 !important;
}

.react-datepicker__day--selected {
  color: white !important;
  background-color: rgb(139 92 246) !important;
}

.react-datepicker__day--disabled {
  color: #ccc !important;
}

.react-datepicker__current-month, .react-datepicker-time__header {
  color: #555555 !important;
  font-weight: 500 !important;
}

.react-datepicker__time-container .react-datepicker__time .react-datepicker__time-box ul.react-datepicker__time-list li.react-datepicker__time-list-item {
  color: #6b6b6b !important;
}

.react-datepicker__time-container .react-datepicker__time .react-datepicker__time-box ul.react-datepicker__time-list li.react-datepicker__time-list-item--selected {
  color: white !important;
  background-color: rgb(139 92 246) !important;
}

.react-datepicker__time-container {
  border-radius: 17px !important;
  border: 1px solid #d8d8d8 !important;
  margin-left: 3px;
}

.react-datepicker__time-container .react-datepicker__time {
  border-bottom-right-radius: 17px !important;
  border-bottom-left-radius: 17px !important;
}
