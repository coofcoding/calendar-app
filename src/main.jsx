import React from 'react'
import { Provider } from 'react-redux';
import ReactDOM from 'react-dom/client'
import { CalendarApp } from './CalendarApp';
import { BrowserRouter } from 'react-router-dom';

// FONTS & STYLES
import './index.css'
import '@fontsource/poppins/100.css';
import '@fontsource/poppins/200.css';
import '@fontsource/poppins/300.css';
import '@fontsource/poppins/400.css';
import '@fontsource/poppins/500.css';
import '@fontsource/poppins/600.css';
import '@fontsource/poppins/700.css';
import '@fontsource/poppins/800.css';
import '@fontsource/poppins/900.css';
import { store } from './store';

ReactDOM.createRoot(document.getElementById('root')).render(
  // <React.StrictMode>
  <Provider store={ store }>
    <BrowserRouter>
      <CalendarApp />
    </BrowserRouter>
  </Provider>
  // </React.StrictMode>,
)
