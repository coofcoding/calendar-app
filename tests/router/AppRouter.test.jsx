import { render, screen } from "@testing-library/react";
import { AppRouter } from "../../src/router/AppRouter";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import { useAuthStore } from "../../src/hooks";
import { store } from "../../src/store";

jest.mock("./../../src/hooks/useAuthStore");


jest.mock("react-modal", () => ({
    ...jest.requireActual("react-modal"),
    setAppElement: () => {},
}));

describe("Pruebas en <AppRouter />", () => {
    
    const mockCheckAuthToken = jest.fn();

    beforeEach( () => jest.clearAllMocks() ); 

    test("debe de mostrar la pantalla de carga y llamar checkAuthToken", () => {

        useAuthStore.mockReturnValue({
            status: 'checking',
            checkAuthToken: mockCheckAuthToken
        });

        render(
            <Provider store={store}>
                <MemoryRouter>
                    <AppRouter />
                </MemoryRouter>
            </Provider>
        );

        expect( screen.getByText('Loading...') ).toBeTruthy();

        expect( mockCheckAuthToken ).toHaveBeenCalled();
    });
});
