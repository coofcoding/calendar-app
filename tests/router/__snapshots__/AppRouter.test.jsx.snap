// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter /> debe de mostrar el login en caso de no estar autenticado 1`] = `
<div>
  <div
    class="w-full md:grid md:grid-flow-col h-screen min-h-screen max-h-screen"
  >
    <div
      class="p-10 shadow-xl grid content-center min-h-full bg-white bg-[radial-gradient(#e5e7eb_1px,transparent_1px)] [background-size:16px_16px]"
    >
      <h3
        class="font-bold text-violet-700 text-4xl mb-6"
      >
        Sign in
      </h3>
      <form>
        <div
          class="my-2"
        >
          <input
            class="px-3 h-11 focus:bg-white w-full text-slate-600 rounded border placeholder-slate-400 spacing font-medium text-sm outline-none focus:ring ring-violet-300 focus:border-violet-500 transition-all duration-150 hover:border-violet-400 bg-slate-100 border-slate-300 py-2"
            name="loginEmail"
            placeholder="email"
            type="text"
            value=""
          />
        </div>
        <div
          class="my-2"
        >
          <input
            class="px-3 h-11 focus:bg-white w-full text-slate-600 rounded border placeholder-slate-400 spacing font-medium text-sm outline-none focus:ring ring-violet-300 focus:border-violet-500 transition-all duration-150 hover:border-violet-400 bg-slate-100 border-slate-300 py-2"
            name="loginPassword"
            placeholder="password"
            type="password"
            value=""
          />
        </div>
        <div
          class="my-2"
        >
          <input
            class="bg-gradient-to-r from-violet-600 to-violet-700 hover:scale-[1.02] transition-all duration-150  w-full h-12 rounded font-bold text-white cursor-pointer"
            type="submit"
            value="sign in"
          />
        </div>
      </form>
    </div>
    <div
      class="p-10 grid content-center min-h-full [background:radial-gradient(125%_125%_at_50%_10%,#100037_40%,#63e_100%)]"
    >
      <h3
        class="font-bold text-white text-4xl mb-6"
      >
        Sign up
      </h3>
      <form>
        <div
          class="my-2"
        >
          <input
            class="px-3 h-11 focus:bg-white w-full text-slate-600 rounded border placeholder-slate-400 spacing font-medium text-sm outline-none focus:ring ring-violet-300 focus:border-violet-500 transition-all duration-150 hover:border-violet-400 bg-slate-100 border-slate-300 py-2"
            name="registerName"
            placeholder="name"
            type="text"
            value=""
          />
        </div>
        <div
          class="my-2"
        >
          <input
            class="px-3 h-11 focus:bg-white w-full text-slate-600 rounded border placeholder-slate-400 spacing font-medium text-sm outline-none focus:ring ring-violet-300 focus:border-violet-500 transition-all duration-150 hover:border-violet-400 bg-slate-100 border-slate-300 py-2"
            name="registerEmail"
            placeholder="email"
            type="email"
            value=""
          />
        </div>
        <div
          class="my-2"
        >
          <input
            class="px-3 h-11 focus:bg-white w-full text-slate-600 rounded border placeholder-slate-400 spacing font-medium text-sm outline-none focus:ring ring-violet-300 focus:border-violet-500 transition-all duration-150 hover:border-violet-400 bg-slate-100 border-slate-300 py-2"
            name="registerPassword"
            placeholder="password"
            type="password"
            value=""
          />
        </div>
        <div
          class="my-2"
        >
          <input
            class="px-3 h-11 focus:bg-white w-full text-slate-600 rounded border placeholder-slate-400 spacing font-medium text-sm outline-none focus:ring ring-violet-300 focus:border-violet-500 transition-all duration-150 hover:border-violet-400 bg-slate-100 border-slate-300 py-2"
            name="registerConfirmPassword"
            placeholder="repeat your password"
            type="password"
            value=""
          />
        </div>
        <div
          class="my-2"
        >
          <input
            class="bg-gradient-to-r from-violet-600 to-violet-700 hover:scale-[1.02] transition-all duration-150  w-full h-12 rounded font-bold text-white cursor-pointer"
            type="submit"
            value="sign up"
          />
        </div>
      </form>
    </div>
  </div>
</div>
`;
